scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -186,57 +185,44 @@ object ParquetAvroIO {
     conf: Configuration = ReadParam.DefaultConfiguration
   ) {
     val avroClass: Class[A] = ScioUtil.classOf[A]
-    val readSchema: Schema = {
-      if (classOf[SpecificRecordBase] isAssignableFrom avroClass) {
-        ReflectData.get().getSchema(avroClass)
-      } else {
-        projection
-      }
-    }
-
-    def read(sc: ScioContext, path: String): HadoopFormatIO.Read[JBoolean, T] = {
-      val job = Job.getInstance(conf)
-      GcsConnectorUtil.setInputPaths(sc, job, path)
-      job.setInputFormatClass(classOf[AvroParquetInputFormat[T]])
-      job.getConfiguration.setClass("key.class", classOf[Void], classOf[Void])
-      job.getConfiguration.setClass("value.class", avroClass, avroClass)
+    val isSpecific: Boolean = classOf[SpecificRecordBase] isAssignableFrom avroClass
+    val readSchema: Schema =
+      if (isSpecific) ReflectData.get().getSchema(avroClass) else projection
 
+    def read(sc: ScioContext, path: String): SCollection[T] = {
+      val hadoopConf = new Configuration(conf)
       // Needed to make GenericRecord read by parquet-avro work with Beam's
       // org.apache.beam.sdk.coders.AvroCoder.
-      if (ScioUtil.classOf[A] == classOf[GenericRecord]) {
-        job.getConfiguration.setBoolean("parquet.avro.compatible", false)
-        job.getConfiguration.set(
-          "parquet.avro.data.supplier",
+      if (!isSpecific) {
+        hadoopConf.setBoolean(AvroReadSupport.AVRO_COMPATIBILITY, false)
+        hadoopConf.set(
+          AvroReadSupport.AVRO_DATA_SUPPLIER,
           "org.apache.parquet.avro.GenericDataSupplier"
```

Comment:
I would maybe replace this to make it a static dependency checked by the compiler.
```suggestion
          classOf[org.apache.parquet.avro.GenericDataSupplier].getCanonicalName
```